### PR TITLE
dev-python/peewee: version bump to 3.1.3

### PR DESCRIPTION
**Pull request by lluixhi**
Introduce "native-extensions" use flag for the cython-based speedups.

The 2.x/3.x version change breaks API, should it be in a different slot?

### CODE REVIEW

**Comment by mgorny** [comment]
Can 2.x be installed in parallel to 3.x?

Also, do you happen to be using some kind of revdep of this?

**Comment by lluixhi** [comment]
oof. I just realized this update will break flask-peewee, wtf-peewee, and flask-admin -- should I add commits fixing their dependencies?

**Comment by mgorny** [comment]
Yes, that would be helpful. Remember to revbump the newest versions.

**Comment by lluixhi** [comment]
@mgorny Does that look good or was I being a bit overzealous? I suppose either way you can pick the commits you want to merge.

**Comment by lluixhi** [comment]
wait hold up, don't merge this yet -- noticed another mistake..

File: dev-python/peewee/peewee-3.0.18.ebuild [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+inherit distutils-r1
+
+DESCRIPTION="Small python ORM"
+HOMEPAGE="https://github.com/coleifer/peewee/"
+SRC_URI="https://github.com/coleifer/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc examples native-extensions"
```
[the remark this answers is not in the document]
Comment (lluixhi) [reply dev-python/peewee/peewee-3.0.18.ebuild]: A while ago, the native extensions were experimental, but I think that's changed as of 3.x, so okay.

File: dev-python/peewee/peewee-3.0.18.ebuild [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	native-extensions? (
+		>=dev-python/cython-0.22.1[${PYTHON_USEDEP}]
+		dev-db/sqlite:3
+	)
+	doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )"
+# Req'd to ensure a unique tmp.db for each python impl running the testsuite.
+DISTUTILS_IN_SOURCE_BUILD=1
```
[the remark this answers is not in the document]
Comment (lluixhi) [reply dev-python/peewee/peewee-3.0.18.ebuild]: Hmmm, yeah the file isn't called `tmp.db` it's called `peewee_test.db` now -- `runtests.py` deletes the database once it's done running, so it's probably unnecessary?

File: dev-python/flask-peewee/flask-peewee-3.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
[the remark this answers is not in the document]
Comment (lluixhi) [reply dev-python/flask-peewee/flask-peewee-3.0.0.ebuild]: Yep

File: dev-python/flask-peewee/flask-peewee-3.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+
+inherit distutils-r1
+
+RESTRICT="test" # broken
```
[the remark this answers is not in the document]
Comment (lluixhi) [reply dev-python/flask-peewee/flask-peewee-3.0.0.ebuild]: alright

File: dev-python/wtf-peewee/wtf-peewee-3.0.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+EAPI=6
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+
+inherit distutils-r1
+
+DESCRIPTION="WTForms integration for peewee models"
+HOMEPAGE="https://github.com/coleifer/wtf-peewee/"
+SRC_URI="https://github.com/coleifer/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="BSD"
```
[the remark this answers is not in the document]
Comment (lluixhi) [reply dev-python/wtf-peewee/wtf-peewee-3.0.0.ebuild]: the maintainer already bumped wtf-peewee to 3.0.0 so I'm just going to leave it alone.

File: dev-python/flask-peewee/flask-peewee-3.0.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	dev-python/jinja[${PYTHON_USEDEP}]
+	>=dev-python/peewee-3.0.0[${PYTHON_USEDEP}]
+	dev-python/wtforms[${PYTHON_USEDEP}]
+	>=dev-python/wtf-peewee-3.0.0[${PYTHON_USEDEP}]"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/nose[${PYTHON_USEDEP}] )
+	doc? ( dev-python/sphinx[${PYTHON_USEDEP}]
+	)"
```
[the remark this answers is not in the document]
Comment (lluixhi) [reply dev-python/flask-peewee/flask-peewee-3.0.0.ebuild]: sure

**Comment by lluixhi** [comment]
@mgorny 
I think that's everything?

Bumping to peewee to 3.1.2 now because a new version came out.

**Comment by lluixhi** [comment]
... on second thought I don't think im going to touch `flask-admin` because I think the dependencies are already messed up and I can't run the testsuite right now.

File: dev-python/peewee/peewee-3.0.18.ebuild (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgorny) [reply dev-python/peewee/peewee-3.0.18.ebuild]: Yes, sounds like it.

**Comment by lluixhi** [comment]
@patricklauer @mgorny 
Can this be merged or is something else wrong?

File: dev-python/peewee/peewee-3.1.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+	distutils-r1_python_compile
+}
+
+python_compile_all() {
+	use doc && emake -C docs html
+}
+
+python_test() {
+	# Testsuite run using runtests.py does not require deps listed in previous ebuild
+	"${PYTHON}" ./runtests.py || die "tests failed under ${EPYTHON}"
```
Comment (mgorny) [inline dev-python/peewee/peewee-3.1.3.ebuild]: EPYTHON, please.

File: dev-python/flask-peewee/flask-peewee-0.6.5-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (mgorny) [inline dev-python/flask-peewee/flask-peewee-0.6.5-r1.ebuild]: Do the usual `git mv` here, and retain stable keywords. This is only a dep update, right?

File: dev-python/flask-peewee/flask-peewee-0.6.7-r1.ebuild [hunk 9]
```diff
@@ -1,8 +1,8 @@
-# Copyright 1999-2015 Gentoo Foundation
+# Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=5
-PYTHON_COMPAT=( python2_7 )
+PYTHON_COMPAT=( python2_7 python3_{4,5} )
```
Comment (mgorny) [inline dev-python/flask-peewee/flask-peewee-0.6.7-r1.ebuild]: That doesn't look like 'fixing deps' to me. Really belongs in a separate commit (to the same revision).

**Comment by flappyports** [comment]
no activity
